grading/main_compile.cpp

```diff
@@ -72,108 +72,102 @@ int main(int argc, char *argv[]) {
   std::cout << "Compiling User Code..." << std::endl;
 
   system("find . -type f -exec ls -sh {} +");
+  assert (config_json.find("testcases") != config_json.end());
 
-  // Run each COMPILATION TEST
-  nlohmann::json::iterator tc = config_json.find("testcases");
-  assert (tc != config_json.end());
-  for (unsigned int i = 1; i <= tc->size(); i++) {
```

Comment:
Why does this loop go away?
We can still have multiple compilation test cases.
We still need a loop.

Reply:
Is this because main_compile will never do "all" since we got rid of the -1 option?

Reply:
The cpp scripts used to be able to be called so that the loop was run internally, rather than handled by the python code. There is no place left in the system that still wants the cpp code to run all testcases, and it was only helpful when the script was being manually run by a dev.